### PR TITLE
+ frozen_error_class

### PR DESCRIPTION
**Pull request by marcandre**
Ruby 2.5 will now raise `FrozenError` instead of `RuntimeError`. It is a descendant, so I assume most spec will pass (for now). Still, it would probably be good to add the following helper so it is easy to make the specs more precise and still be backwards compatible.

Please note that for `backports`, not doing something like this would be a huge hastle. `backports` makes no attempts to change the behavior of older version in cases like this, but always tests future specs against previous versions (with backported code). If the specs are updated without a way for `backports` to specify that `RuntimeError` is actually ok, I'll have to add tags for a huge number of specs, just because of the class change. I'm not sure I'm being clear though lol

@eregon thoughts on the matter?

### CODE REVIEW

File: lib/mspec/helpers/frozen_error_class.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
+# This helper makes it easy to write version independent
+# specs for frozen objects.
+unless respond_to? :frozen_error_class
+  if VersionGuard.new("2.5").match?
```
Comment (eregon) [inline lib/mspec/helpers/frozen_error_class.rb]: Using `ruby_version_is` might be nicer here.

**Comment by eregon** [comment]
Looks good, thanks!

File: lib/mspec/helpers/frozen_error_class.rb [hunk 2]
```diff
@@ -0,0 +1,13 @@
+# This helper makes it easy to write version independent
+# specs for frozen objects.
+unless respond_to? :frozen_error_class
```
Comment (eregon) [inline lib/mspec/helpers/frozen_error_class.rb]: Do you override it in Backports or so?

File: lib/mspec/helpers/frozen_error_class.rb (hunk 1 above)
Comment (eregon) [reply lib/mspec/helpers/frozen_error_class.rb]: 19d929eb0

File: lib/mspec/helpers/frozen_error_class.rb (hunk 2 above)
Comment (marcandre) [reply lib/mspec/helpers/frozen_error_class.rb]: That would be the plan, yes.

**Pull request merged**